src/Raven.Server/Documents/ETL/ExtractedItem.cs

```diff
@@ -9,18 +11,38 @@ protected ExtractedItem()
             
         }
 
-        protected ExtractedItem(Document document)
+        protected ExtractedItem(Document document, string collection)
         {
             DocumentKey = document.Key;
             Etag = document.Etag;
             Document = document;
+            Collection = collection;
+
+            if (collection == null)
+            {
+                CalculatedCollectionName = new Lazy<LazyStringValue>(() =>
+                {
```

Comment:
I made it a lazy because SQL ETL doesn't make use of it so it won't be evaluated then. However since that is needed only for Raven ETL I'll move it to from base `ExtractedItem` class to `RavenEtlItem` and drop `Lazy` wrapper